Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs

```diff
@@ -1069,7 +1069,21 @@ public abstract class Container
         /// <returns>A new instance of <see cref="TransactionalBatch"/>.</returns>
         public abstract TransactionalBatch CreateTransactionalBatch(PartitionKey partitionKey);
 
-#if PREVIEW
+#if PREVIEW        
+        /// <summary>
+        /// Deletes all items in the Container with the specified <see cref="PartitionKey"/> value.
+        /// </summary>
+        /// <param name="partitionKey">The <see cref="PartitionKey"/> of the items to be deleted.</param>
+        /// <param name="requestOptions">(Optional) The options for the Partition Key Delete request.</param>
+        /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
+        /// <returns>
+        /// A <see cref="Task"/> containing a <see cref="ResponseMessage"/> which wraps a <see cref="Stream"/>.
+        /// </returns>
+        public abstract Task<ResponseMessage> DeleteAllItemsByPartitionKeyAsync(
```

Comment:
Why is there not a typed response version?
```suggestion
        public abstract Task<ResponseMessage> DeleteAllItemsByPartitionKeyStreamAsync(
```

Reply:
Shouldn't it take a Stream if we are naming it so?

Reply:
Not necessarily, for example there is a CreateDatabaseStreamAsync that already does this.

Reply:
Renamed to DeleteAllItemsByPartitionKeyStreamAsync